server/src/main/java/org/eclipse/openvsx/LocalRegistryService.java

```diff
@@ -313,9 +324,20 @@ public ResultJson createNamespace(NamespaceJson json, String tokenValue) {
         if (namespaceIssue.isPresent()) {
             throw new ErrorResultException(namespaceIssue.get().toString());
         }
-        var token = users.useAccessToken(tokenValue);
-        if (token == null) {
-            throw new ErrorResultException("Invalid access token.");
+        if (tokenValue == null) {
+            var principal = users.getOAuth2Principal();
+            if (principal == null) {
+                throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+            }
+            var user = users.updateUser(principal);
+            if(!user.getRole().equals("admin")) {
+                throw new ErrorResultException("Not Authorized.");
```

Comment:
`ErrorResultException` now also accepts a `HttpStatus` as second parameter. This should be `FORBIDDEN` like above (same for the next exception below).